lib/order.rb

```diff
@@ -0,0 +1,34 @@
+class Order 
+    attr_reader :menu, :basket
+    
+    def initialize(menu = Menu.new) 
+    @menu = menu
+    @basket = []
+    end
+    
+    def view
+      @menu.dishes.each do | item, price |
+        puts "#{item}: £#{price}"
+      end   
+    end
+
+    def add(dish, quantity)
+        total_price = 0
+        @menu.dishes.each do | item, price |
+            if dish == item
+                total_price = price * quantity
+            end
+        end
+        fail 'Sorry this dish is not available' if total_price == 0
+        @basket.push ([dish, quantity, total_price])
+    end
```

Comment:
Could this be split into multiple methods? remember to keep your methods simple!